### ISSUE TITLE
compress_by_redundancy() Indexing issue

### ISSUE THREAD

**Issue by alphatangojuliett** [RadioAstronomySoftwareGroup/pyuvdata#935]
### When trying to compress a UVData object containing a subset of data from even one .uvh5 file, I receive the following error:

   _5595                         dtype=np.int,
   5596                     )
-> 5597                     conj_inds = np.array(conj_group_inds[np.array(conj_orientation)])
   5598                     # check that the integration times are all the same
   5599                     int_times = np.concatenate(

IndexError: index 78 is out of bounds for axis 0 with size 24

**### 
Minimal code to reproduce issue, where 'file' here is, as an example, zen.2459100.42005.sum.uvh5 (link: https://drive.google.com/file/d/1SYYI0ys91AfRRxHuQCFruehvMuXgD5-1/view?usp=sharing)**

_uvd=UVData()
uvd.read(file, axis='blt', keep_all_metadata=False,frequencies=frequencies,polarizations=pol)
uvd_compress=uvd.compress_by_redundancy(method='average', tol=1.0, inplace=False, keep_all_metadata=False)_

**Comment by bhazelton** [comment]
@alphatangojuliett what are you specifying for pol, frequencies here? Have you tried this with just a simple read like:
```uvd.read(file)```?

**Comment by bhazelton** [comment]
@alphatangojuliett We have reproduced your error and we think we found the bug. We're still working on fully testing it, but there's a fix that we think is the right fix in the `fix_compress_average` branch if you'd like to try it out. I am confident it won't error, but we still need to implement better testing to ensure it is always doing the right thing.

We went back to our tests to figure out why we hadn't caught this before and found that our test file was very special -- all the baseline groups it had were either all not conjugated or all conjugated, there were none with a mix. This bug only shows up with a mixed set.

Thank you for finding this bug for us!
